internal/incident/incident.go

```diff
@@ -533,6 +481,74 @@ func (i *Incident) RestoreEscalationStateRules(states []*EscalationState) {
 	}
 }
 
+// getEscalationRecipientsChannel returns the configured channel of all the provided escalation recipients.
+func (i *Incident) getEscalationRecipientsChannel(escalation *rule.Escalation, t time.Time) recipientsChannelType {
+	if i.Recipients == nil {
+		i.Recipients = make(map[recipient.Key]*RecipientState)
+	}
+
+	managed := i.HasManager()
+	contactChannels := make(recipientsChannelType)
+	for _, escalationRecipient := range escalation.Recipients {
+		state := i.Recipients[escalationRecipient.Key]
+		if state == nil {
+			continue
+		}
+
+		if !managed || state.Role > RoleRecipient {
+			for _, c := range escalationRecipient.Recipient.GetContactsAt(t) {
+				if contactChannels[c] == nil {
+					contactChannels[c] = make(map[string]struct{})
+				}
+				if escalationRecipient.ChannelType.Valid {
+					contactChannels[c][escalationRecipient.ChannelType.String] = struct{}{}
+				} else {
+					contactChannels[c][c.DefaultChannel] = struct{}{}
+				}
+			}
+		}
+	}
+
+	return contactChannels
+}
+
+// getRecipientsChannel returns all the configured channels of the current incident and escalation recipients.
+func (i *Incident) getRecipientsChannel(t time.Time) recipientsChannelType {
+	if i.Recipients == nil {
+		i.Recipients = make(map[recipient.Key]*RecipientState)
+	}
+
+	managed := i.HasManager()
+	contactChannels := make(recipientsChannelType)
+	escalationRecipients := make(map[recipient.Key]bool)
+	for escalationID := range i.EscalationState {
+		escalation := i.runtimeConfig.GetRuleEscalation(escalationID)
+		for c, channels := range i.getEscalationRecipientsChannel(escalation, t) {
+			escalationRecipients[recipient.ToKey(c)] = true
+			contactChannels[c] = channels
```

Comment:
Wouldn't you want to merge `channels` into `contactChannels[c]` if the latter already contains something? (Maybe it would also make sense to change `getEscalationRecipientsChannel()` so that it adds everything to a map given as an input instead of returning a new map.)